drake/examples/Pendulum/Pendulum.h

```diff
@@ -3,100 +3,137 @@
 
 #include <iostream>
 #include <cmath>
-#include "DrakeSystem.h"
-#include "LCMCoordinateFrame.h"
-#include "BotVisualizer.h"
+#include "System.h"
+#include "LQR.h"
 
-class Pendulum : public DrakeSystem {
+using namespace std;
+
+template <typename ScalarType = double>
+class PendulumState  { // models the Drake::Vector concept
 public:
-  Pendulum(const std::shared_ptr<lcm::LCM>& lcm) :
-          DrakeSystem("Pendulum",
-                  std::make_shared<CoordinateFrame>("PendulumContState",std::vector<std::string>({"theta","thetadot"})),
-                  nullptr,
-                  std::make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumInput",std::vector<std::string>({"tau"}),lcm),
-                  std::make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumState",std::vector<std::string>({"theta","thetadot"}),lcm)),
-          m(1.0), // kg
-          l(.5),  // m
-          b(0.1), // kg m^2 /s
-          lc(.5), // m
-          I(.25), // m*l^2; % kg*m^2
-          g(9.81) // m/s^2
-  {}
-  virtual ~Pendulum(void) {};
+  PendulumState(void) : theta(0), thetadot(0) {};
+  PendulumState(const Eigen::Matrix<ScalarType,2,1>& x) : theta(x(0)), thetadot(x(1)) {};
 
-  virtual Eigen::VectorXd dynamics(double t, const Eigen::VectorXd& x, const Eigen::VectorXd& u) const override {
-    return dynamics_implementation(t,x,u);
-  }
-  virtual Drake::TaylorVecX dynamics(Drake::TaylorVarX t, const Drake::TaylorVecX& x, const Drake::TaylorVecX& u) const override {
-    return dynamics_implementation(t,x,u);
+  PendulumState& operator=(const Eigen::Matrix<ScalarType,2,1>& x) {
```

Comment:
I don't think it's necessary to have cast operators to different Eigen types. Just return the smallest type you know will fit the state vector (in this case a Vector2d, in some cases it will need to be a VectorXd). If you need it to be a VectorXd, just construct a VectorXd using the Vector2d that comes out.